library/public/types.h

```diff
@@ -58,6 +71,56 @@ class F3D_EXPORT ratio_t
   double Value;
 };
 
+/**
+ * Describe a RGB color.
+ */
+class F3D_EXPORT color_t
+{
+public:
+  color_t() = default;
+  color_t(const std::vector<double>& vec)
+  {
+    if (vec.size() != 3)
+    {
+      throw f3d::type_construction_exception("Provided vector does not have the right size");
+    }
+    std::copy_n(vec.begin(), 3, this->RGB.begin());
+  }
+  color_t(double red, double green, double blue)
+  {
+    this->RGB[0] = red;
+    this->RGB[1] = green;
+    this->RGB[2] = blue;
+  }
+  color_t(const std::initializer_list<double>& list)
+  {
+    if (list.size() != 3)
+    {
+      throw f3d::type_construction_exception("Provided list does not have the right size");
+    }
+    std::copy_n(list.begin(), 3, this->RGB.begin());
+  }
+  operator std::vector<double>() const
+  {
+    return std::vector<double>(this->RGB.begin(), this->RGB.end());
+  }
+  bool operator==(const color_t& other) const
+  {
+    return this->RGB == other.RGB;
+  }
+  bool operator!=(const color_t& other) const
+  {
+    return this->RGB != other.RGB;
+  }
+  const double* data() const
+  {
+    return this->RGB.data();
+  }
+
+private:
+  std::array<double, 3> RGB;
```

Comment:
I guess we could consider templating this and declaring rgba_color_t and rgb_color_t. There is no usecases for now.

Reply:
Yes, we'll see later